Content.Server/EntityEffects/EntityEffectSystem.cs

```diff
@@ -0,0 +1,968 @@
+using Content.Server.Zombies;
+using Content.Shared.Atmos;
+using Content.Shared.Audio;
+using Content.Shared.Coordinates.Helpers;
+using Content.Shared.EntityEffects.EffectConditions;
+using Content.Shared.EntityEffects.Effects.PlantMetabolism;
+using Content.Shared.EntityEffects.Effects.StatusEffects;
+using Content.Shared.EntityEffects.Effects;
+using Content.Shared.EntityEffects;
+using Content.Shared.Maps;
+using Content.Shared.Mind.Components;
+using Content.Shared.Popups;
+using Content.Shared.Random;
+using Content.Shared.Zombies;
+using Robust.Server.GameObjects;
+using Robust.Shared.Audio;
+using Robust.Shared.Audio.Systems;
+using Robust.Shared.GameObjects;
+using Robust.Shared.Map;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+
+using TemperatureCondition = Content.Shared.EntityEffects.EffectConditions.Temperature; // disambiguate the namespace
+using PolymorphEffect = Content.Shared.EntityEffects.Effects.Polymorph;
+
+namespace Content.Server.EntityEffects;
+
+public sealed class EntityEffectSystem : EntitySystem
+{
+    [Dependency] private readonly AtmosphereSystem _atmosphere = default!;
+    [Dependency] private readonly BloodstreamSystem _bloodstream = default!;
+    [Dependency] private readonly ChatSystem _chat = default!;
+    [Dependency] private readonly EmpSystem _emp = default!;
+    [Dependency] private readonly ExplosionSystem _explosion = default!;
+    [Dependency] private readonly FlammableSystem _flammable = default!;
+    [Dependency] private readonly FlashSystem _flash = default!;
+    [Dependency] private readonly IMapManager _mapManager = default!;
+    [Dependency] private readonly IPrototypeManager _protoManager = default!;
+    [Dependency] private readonly IRobustRandom _random = default!;
+    [Dependency] private readonly SharedMapSystem _map = default!;
+    [Dependency] private readonly MutationSystem _mutation = default!;
+    [Dependency] private readonly NarcolepsySystem _narcolepsy = default!;
+    [Dependency] private readonly PlantHolderSystem _plantHolder = default!;
+    [Dependency] private readonly PolymorphSystem _polymorph = default!;
+    [Dependency] private readonly RespiratorSystem _respirator = default!;
+    [Dependency] private readonly SharedAudioSystem _audio = default!;
+    [Dependency] private readonly SharedPointLightSystem _pointLight = default!;
+    [Dependency] private readonly SharedPopupSystem _popup = default!;
+    [Dependency] private readonly SmokeSystem _smoke = default!;
+    [Dependency] private readonly SpreaderSystem _spreader = default!;
+    [Dependency] private readonly TemperatureSystem _temperature = default!;
+    [Dependency] private readonly SharedTransformSystem _xform = default!;
+    [Dependency] private readonly VomitSystem _vomit = default!;
+
+    public override void Initialize()
+    {
+        base.Initialize();
+
+        SubscribeLocalEvent<CheckEntityEffectConditionEvent<TemperatureCondition>>(OnCheckTemperature);
```

Comment:
these being global subs kinda sucks. It would be a lot nicer if it was raised on the source entity and then you could subscribe via a component. It would reduce a lot of the overhead.

Reply:
raised on which component of the source entity?

Reply:
you should just be able to raise it on the entity the effect targets, the same way as we do `RaiseLocalEvent(uid, ref ev);`